src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
 // write code here
+
+const employeeList = document.querySelector('ul');
+
+// Convert the data-salary attribute from a string to a number
+
+function parseSalary(salarySring) {
+  if (!salarySring) {
+    return 0;
+  }
+
+  return parseFloat(salarySring.replace(/[^0-9.-]+/g, ''));
+}
+
+// Take a list element and sorts it by the data-salary attribute
+
+function sortList(list) {
+  const items = Array.from(list.children);
+
+  items.sort((a, b) => {
+    const salaryA = parseSalary(a.getAttribute('data-salary'));
+    const salaryB = parseSalary(b.getAttribute('data-salary'));
+
+    return salaryB - salaryA;
+  });
+
+  items.forEach((item) => list.appendChild(item));
+}
+
+// Returns an array of employee objects with specified schema.
+
+function getEmployees(list) {
+  return Array.from(list.children).map((item) => ({
+    name: item.innerText,
+    position: item.getAttribute('data-position'),
+    salary: parseSalary(item.getAttribute('data-position')),
+    age: parseInt(item.getAttribute('data-position')),
```

Comment:
There is an error here. The `data-position` attribute is being used to retrieve the age, which is incorrect. It should be `data-age`.